circom-prover/src/witness.rs

```diff
@@ -0,0 +1,63 @@
+use num::{BigInt, BigUint};
+use std::{collections::HashMap, str::FromStr};
+
+/// Witness function signature for rust_witness (inputs) -> witness
+type RustWitnessWtnsFn = fn(HashMap<String, Vec<BigInt>>) -> Vec<BigInt>;
+/// Witness function signature for witnesscalc_adapter (inputs, .dat file path) -> witness
+type WitnesscalcWtnsFn = fn(HashMap<String, Vec<BigInt>>, &str) -> Vec<BigInt>;
+
+pub enum WitnessFn {
+    WitnessCalc(WitnesscalcWtnsFn),
+    RustWitness(RustWitnessWtnsFn),
+}
+
+pub enum WitnessLib {
+    WitnessCalc,
+    RustWitness,
+}
+
+/// To create witness functions corresponding to different witness generation libs.
+#[macro_export]
+macro_rules! create_witness_fn {
```

Comment:
I think this macro is a bad abstraction. Why do we want to hide which adapter is being used to create the witness function? e.g. why can't we use `rust_witness` and `witnesscalc_adapter` directly? Adding this adds a layer of complexity, what is the justification?

Reply:
removed in [b22be2c](https://github.com/zkmopro/mopro/pull/304/commits/b22be2cff51e3db57501cb480fbdd60bba92e727)

Reply:
removed in [d47fd38](https://github.com/zkmopro/mopro/pull/304/commits/d47fd38ce6edda89ffbc03479b6abb2b505330d7)